crypto/pkcs7/pk7_ecdsa.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/pkcs7.h>
+#include <openssl/x509.h>
+#include "crypto/asn1.h"
+#include "pk7_local.h"
+
+int pkcs7_ecdsa_dsa_sign(PKCS7_SIGNER_INFO *si, int verify)
```

Comment:
Perhaps pkcs7_ecdsa_dsa_sign should be pkcs7_ecdsa_or_dsa_sign to be clearer?
And actually given it is both sign and verify calling it just "sign" seems wrong.
Perhaps pkcs7_ecdsa_or_dsa_sign_verify?

And looking at all the code we are using the three new functions in only one place - so why are we adding these additional global symbols for a single user and each of these functions is rather short/simple?

Reply:
you could also argue that sign_verify is not a good name either.. They look more like a resolve stage for doing those particular operations..

Reply:
Indeed. "resolve" or "x" or something less misdirecting would be better IMHO - but if only used once then they don't need to be separate functions - or name them for where they are called from ... or "bridge" in the name or ...

Reply:
will collapse into pkcs7_lib.c - It is not a large source file (in comparison to others).